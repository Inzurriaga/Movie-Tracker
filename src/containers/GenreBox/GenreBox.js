import React, { Component } from "react"
import { Link } from "react-router-dom"
import { connect } from "react-redux"


export class GenreBox extends Component {
    constructor() {
        super();
        this.state = {
            genre: ["Action", "Horror", "Comedy"]
        }
    }

    render = () => {
        const genreDisplay = this.props.genres.map((genre, index) => {
            let backgroundImage = {backgroundImage: `url( http://image.tmdb.org/t/p/original${genre.results[0].backdrop_path})`}
            return(
                   <Link key={index} to={`/Movies/genre/${index}`} style={backgroundImage} className="genre-box"><h3>{this.state.genre[index]}</h3></Link>
                 )
        })
        return(
            <div className="Genre-Display">
                {genreDisplay}
            </div>)
    }
}

export const mapStateToProps = (state) => ({
    genres: state.genres
})

export default connect(mapStateToProps)(GenreBox)
