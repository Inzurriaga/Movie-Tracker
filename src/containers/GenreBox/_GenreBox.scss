.Genre-Display {
  display: flex;
  flex-direction: column;
  height: 100%;
  justify-content: space-between;

  .genre-box {
    display: flex;
    position: relative;
    height: 30.5%;
    text-decoration: none;
    align-items: center;
    justify-content: center;
    background-size: contain;
    background-position: top center;
    transition: all .5s ease;
    box-shadow: 0px 2px 10px 3px #00000017;
    cursor: pointer;
    border-radius: 3px;

    &:hover {
      &:before {
        content: "";
        position: absolute;
        width: 100%;
        height: 100%;
        top: 0px;
        background: rgba(35, 31, 32, .6);
      }
    }

    h3 {
      position: relative;
      z-index: 3;
      font-size: 2em;
      letter-spacing: 0.04em;
      text-transform: uppercase;
      text-shadow: 2px 2px 5px black;
      font-family: $font-Abel;
      display: table-cell;
      vertical-align: bottom;
      color: white;
    }
  }
}
