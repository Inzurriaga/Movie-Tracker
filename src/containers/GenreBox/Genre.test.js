import React from 'react';
import { GenreBox, mapStateToProps } from './GenreBox';
import { shallow } from 'enzyme';

const mockGenreProps = [
  {
    page: 1,
    results: [
      {
        adult: false,
        backdrop_path: "/w2PMyoyLU22YvrGK3smVM9fW1jj.jpg",
        genre_ids: [28, 12, 878],
        id: 299537,
        original_title: "Captain Marvel"
      }
    ],
    total_pages: 1176,
    total_results: 23512,
  }
]


describe('GenreBox', () => {
  let grenreBox;

  describe('GenreBox Component', () => {

    beforeEach(() => {
      grenreBox = shallow(<GenreBox genres={mockGenreProps}/>)
    })

    it('should match the snapshot', () => {

      expect(grenreBox).toMatchSnapshot()

    })

    it('should have default properties', () => {

      expect(grenreBox.state()).toEqual({
        genre: ["Action", "Horror", "Comedy"]
      });

    })

  })


  describe('GenreBox mapStateToProps', () => {

    it('should return an object with genres information', () => {
      const mockState = {
        genres: {
                  page: 1,
                  results: [{
                    adult: false,
                    backdrop_path: "/w2PMyoyLU22YvrGK3smVM9fW1jj.jpg",
                    id: 299537,
                    original_language: "en"
                  }],
                  total_pages: 1176,
                  total_results: 23512
                },
        category: []
      }

      const expected = {
        genres: {
                  page: 1,
                  results: [{
                    adult: false,
                    backdrop_path: "/w2PMyoyLU22YvrGK3smVM9fW1jj.jpg",
                    id: 299537,
                    original_language: "en"
                  }],
                  total_pages: 1176,
                  total_results: 23512
                },

      }

      const mappedProps = mapStateToProps(mockState);

      expect(mappedProps).toEqual(expected)
    })

  })

})
