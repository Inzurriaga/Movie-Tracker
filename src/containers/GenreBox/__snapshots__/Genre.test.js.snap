// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenreBox GenreBox Component should match the snapshot 1`] = `
<div
  className="Genre-Display"
>
  <Link
    className="genre-box"
    key="0"
    style={
      Object {
        "backgroundImage": "url( http://image.tmdb.org/t/p/original/w2PMyoyLU22YvrGK3smVM9fW1jj.jpg)",
      }
    }
    to="/Movies/Action/0"
  >
    <h3>
      Action
    </h3>
  </Link>
</div>
`;
